buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy

```diff
@@ -49,21 +53,81 @@ class TestProgressLogger implements AggregatedEventListener {
         progressLogger = factory.newOperation(TestProgressLogger)
         progressLogger.setDescription('Randomized test runner')
         progressLogger.started()
-        progressLogger.progress('Starting JUnit4 with ' + e.getSlaveCount() + ' jvms')
+        progressLogger.progress("Starting JUnit4 with $e.slaveCount jvms")
+    }
+
+    @Subscribe
+    void onTestResult(AggregatedTestResultEvent e) throws IOException {
+        testsCompleted.incrementAndGet()
+        switch (e.status) {
+        case ERROR:
+        case FAILURE:
+            testsFailed.incrementAndGet()
+            break
+        case IGNORED:
+        case IGNORED_ASSUMPTION:
+            testsIgnored.incrementAndGet()
+            break
+        case OK:
+            break
+        default:
+            throw new IllegalArgumentException(
+                "Unknown test status: [$e.status]")
+        }
+        String suiteName = e.description.displayName
+        log(new SuiteInfo(
+            name: suiteName.substring(suiteName.lastIndexOf('.') + 1,
+                    suiteName.length() - 1),
+            slave: e.slave.id,
+        ))
     }
 
     @Subscribe
     void onSuiteResult(AggregatedSuiteResultEvent e) throws IOException {
-        final int suitesCompleted = suitesCompleted.incrementAndGet();
-        final int testsCompleted = testsCompleted.addAndGet(e.getDescription().testCount())
-        final int testsFailed = testsFailed.addAndGet(e.getErrorCount() + e.getFailureCount())
-        final int testsIgnored = testsIgnored.addAndGet(e.getIgnoredCount())
-        Description description = e.getDescription()
-        String suiteName = description.getDisplayName();
-        suiteName = suiteName.substring(suiteName.lastIndexOf('.') + 1);
-        progressLogger.progress('Suites [' + suitesCompleted + '/' + totalSuites + '], Tests [' + testsCompleted + '|' + testsFailed + '|' + testsIgnored + '], ' + suiteName + ' on J' + e.getSlave().id + ' in ' + formatDurationInSeconds(e.getExecutionTime()))
+        suitesCompleted.incrementAndGet();
+        String suiteName = e.description.displayName
+        SuiteInfo suiteInfo = new SuiteInfo(
+            name: suiteName.substring(suiteName.lastIndexOf('.') + 1),
+            slave: e.slave.id,
+            duration: formatDurationInSeconds(e.executionTime),
+        )
+        lastSuite.set(suiteInfo)
+        log(suiteInfo)
+    }
+
+    private void log(SuiteInfo currentSuite) {
+        SuiteInfo lastSuite = this.lastSuite.get()
+        /* Only use the suite from the current event if we haven't completed a
+          suite yet. Using the current suite at all makes the results useful
+          for long suites like RestIT and using the last completed suite most
+          of the time keeps the suite from jumping back and forth. */
+        String suiteInfo
+        if (lastSuite == null) {
+            suiteInfo = ", current suite is $currentSuite"
```

Comment:
I can look into that. It _should_ be fairly easy to log the test name if we don't have a `lastSuite`. Or I can use an option. The neat thing about doing it with a `lastSuite` is that it just happens automatically when you have a test case that it slow and when you start to get more results it switches to the normal mode.